### PR TITLE
first_true/count_true test refactor and code cleanup

### PR DESCRIPTION
**Pull request by SadiinsoSnowfall**

### CODE REVIEW

**Comment by DenisYaroshevskiy** [comment]
What are the code changes? Is it just formatting?

File: test/unit/module/core/scan_test.hpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
```
Comment (DenisYaroshevskiy) [inline test/unit/module/core/scan_test.hpp:7]: neither first true nor count are a san. They might be reductions but this can only be any bool test too

File: test/unit/module/core/first_true.cpp [hunk 2]
```diff
@@ -36,107 +37,42 @@ TTS_CASE("Check eve::first_true return type (edge cases)")
...
-        if (v.get(i) && mask_at(mask, i))
-          return i;
 
-    return std::nullopt;
-  }
-}
-
-template<typename T, typename C>
-void test_first_true(T v, C cx)
+struct ManualFirstTrue
```
Comment (DenisYaroshevskiy) [inline test/unit/module/core/first_true.cpp:41]: PascalCase all over a sudden.

File: test/unit/module/core/first_true.cpp [hunk 3]
```diff
@@ -36,107 +37,42 @@ TTS_CASE("Check eve::first_true return type (edge cases)")
...
+  test_case<ManualFirstTrue>(eve::first_true, false);
 };
 
 TTS_CASE_TPL("Check eve::first_true behavior on scalars", eve::test::scalar::all_types)
 <typename T>(tts::type<T>)
 {
-  test_first_true(eve::logical<T>{true});
-  test_first_true(eve::logical<T>{false});
+  test_case<ManualFirstTrue>(eve::first_true, eve::logical<T>{true});
+  test_case<ManualFirstTrue>(eve::first_true, eve::logical<T>{false});
```
Comment (DenisYaroshevskiy) [inline test/unit/module/core/first_true.cpp:71]: I'd expect these all to be covered together tbh.

**Review by DenisYaroshevskiy — APPROVED** [review]
